### PR TITLE
[ARTS-527] Practitioner connects to other services using discovery server

### PR DESCRIPTION
**Pull request by anatbal**
## Description
- First client implementation of connectivity between services from the practitioner service.
- Approve together with[ this](https://github.com/NDPH-ARTS/mts-trial-deployment-config/pull/49) PR

### Checks
- [x]  Run docker-compose with 'local' profile
- [x] Run docker-compose with 'dev' profile
- [x] Run on Azure test environment

### Changes
- [ARTS-527] - Practitioner using discovery server in order to connect to other services

### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-527]: https://ndph-arts.atlassian.net/browse/ARTS-527

### CODE REVIEW

File: practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/client/RoleServiceClient.java [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 
 @Component
 public class RoleServiceClient extends AbstractEntityServiceClient {
 
-    public RoleServiceClient(final WebClient.Builder webClientBuilder,
-                             @Value("${role.service.uri}") String serviceUrlBase) {
+    @Autowired
+    public RoleServiceClient(WebClient.Builder webClientBuilder) {
+        this(webClientBuilder, "http://role-service");
+    }
```
Comment (LizaShak) [inline practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/client/RoleServiceClient.java]: I think it is better to have the url/service name in discovery, as part of the configuration

File: docker-compose.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@ services:
       EUREKA_CLIENT_SERVICEURL_DEFAULTZONE: "http://discovery-server:8761/eureka/"
       SPRING_CLOUD_CONFIG_LABEL: "main"
       SPRING_CLOUD_CONFIG_URI: "http://config-server:8888"
+      SITE-SERVICE_RIBBON_LISTOFSERVERS: "http://site-service:8083"
```
Comment (eladiw) [inline docker-compose.yml:27]: This is something completely new.
why we need it? is it a must?

Comment (anatbal) [reply docker-compose.yml:27]: We need this only for local use. This disables the @LoadBalanced annotation and replaces it with a defined url (like we need in docker compose or local run e.g). I add a comment and no need to put this env in prod

File: practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/client/RoleServiceClient.java (hunk 1 above)
Comment (anatbal) [reply practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/client/RoleServiceClient.java]: I reverted it to use config file, @eladiw, @tamirkamara  WDYT?

File: docker-compose.yml (hunk 2 above)
Comment (eladiw) [reply docker-compose.yml:28]: can it be removed? because you have it in the -local.yml file

Comment (anatbal) [reply docker-compose.yml:28]: In the local.yml I put "localhost:8082" for example if someone would run it from the inteliji, and for docker compose we need the service name + port

**Review by eladiw — APPROVED** [review]
LGTM. nice work

File: practitioner-service/src/main/resources/bootstrap-local.yml [hunk 3]
```diff
@@ -5,4 +5,8 @@ spring:
       discovery:
         enabled: false
     discovery:
-      enabled: false
\ No newline at end of file
+      enabled: false
+
+ribbon:
```
Comment (tamirkamara) [inline practitioner-service/src/main/resources/bootstrap-local.yml:10]: Lets add a comment here to explain what it does.

File: practitioner-service/src/main/resources/application-local.yml [hunk 4]
```diff
@@ -26,3 +26,12 @@ config:
   - name: prefix
     displayName: Prefix
     validationRegex: "^[a-zA-Z]{0,35}$"
+
+---
+site-service:
+  ribbon:
+    listOfServers: localhost:8083
```
Comment (tamirkamara) [inline practitioner-service/src/main/resources/application-local.yml:33]: We can simplify the docker-compose if we use the same value it already inject like this example:
_${site.service.uri:http://localhost:8083}_

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: practitioner-service/src/main/resources/bootstrap-local.yml (hunk 3 above)
Comment (anatbal) [reply practitioner-service/src/main/resources/bootstrap-local.yml:10]: Done

File: practitioner-service/src/main/resources/application-local.yml (hunk 4 above)
Comment (anatbal) [reply practitioner-service/src/main/resources/application-local.yml]: Done

**Review by eladiw — APPROVED** [review]

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**